### PR TITLE
coloumn Chart Created

### PR DESCRIPTION
**Pull request by rahuldutta2013**
Please Check

### CODE REVIEW

**Comment by rahuldutta2013** [comment]
tooltip created

File: js/axis.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        var canvas = document.getElementById('myCanvas'),
+            ctx = canvas.getContext('2d');
+        this.canvasWidth = canvasWidth;
+        this.canvasHeight = canvasHeight;
+        var arrOfValue = [],
+            graphHeight = (this.canvasHeight * 97) / 100,
+            origDist = (this.canvasWidth * 10) / 100;
+
+        //craeting the X axis (0,500)
+        ctx.beginPath();
```
Comment (tub99) [inline js/axis.js]: move canvas related logic to a separate class. For drawing a line create Line.js -> function Line(x1,y1,x2,y2)

File: index.html [hunk 2]
```diff
@@ -4,11 +4,14 @@
         <link rel="stylesheet" href="css/canvas.css">
      </head>   
 <body>
-    <canvas id="myCanvas" height="500" width="1000"></canvas>
+    <canvas id="myCanvas" ></canvas>
+     <span class="tooltip"></span>
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
```
Comment (tub99) [inline index.html]: Pull from node_modules

File: index.html [hunk 3]
```diff
@@ -4,11 +4,14 @@
...
     <script src="js/parserManager.js"></script>
```
Comment (tub99) [inline index.html]: Can be grouped to a single js file with gulp/grunt or any other tools(To be done later)

File: js/axis.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
+function Axis(canvasId, canvasHeight, canvasWidth, valOfY) {
+
+    this.axisCalculation = function () {
+        var canvas = document.getElementById('myCanvas'),
```
Comment (tub99) [inline js/axis.js]: Canvas can not be part of axisCalculation

File: js/axis.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            minVal -= limitLow;
+        }
+
+        var gap = (maxVal - (minVal)) / 10;
+
+
+        //creating the division for coloumnChart
+        for (var j = 0; j <= 10; j++) {
+           
+            ctx.moveTo(0, yaxis);
```
Comment (tub99) [inline js/axis.js]: No ctx to be used in axis-calcualtion.. Call the Line constructor to draw a line

File: js/axis.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            x,
+            noOfBar = valOfY.data.length,
+            rightGap = (this.canvasWidth * 5) / 100,
+            gapPlusWidthOfBar = ((this.canvasWidth - rightGap) / noOfBar);
+
+        gapInXAxix = x = widthOfBar = gapPlusWidthOfBar / 2;
+
+     
+        for (var p = 0; p < valOfY.data.length; p++) {
+            ctx.fillStyle  = '#00F';
```
Comment (tub99) [inline js/axis.js]: These properties should be configurable

File: js/chart.js [hunk 7]
```diff
@@ -1,22 +1,87 @@
...
+        return 'yet not decided';
     }
 }
 
-function ColoumnChart(){
+var arrOfOrdinate = [],
+    arrOfVal = [];
+
+
+function ColoumnChart() {
```
Comment (tub99) [inline js/chart.js]: Should be in a different file called columnChart.js

File: js/chart.js [hunk 8]
```diff
@@ -1,22 +1,87 @@
...
     Chart.call(this);
-    this.coloumnChart = function(){
-        return 'coloumn chart generated';
+    this.coloumnChart = function (valOfY, canvasWidth, canvasHeight) { //overriding the coloumnChart property of its parent class
```
Comment (tub99) [inline js/chart.js]: columnChart -> function name should be meaningful

File: js/chart.js [hunk 9]
```diff
@@ -1,22 +1,87 @@
...
+            noOfBar = valOfY.length,
+            rightGap = (canvasWidth * 5) / 100,
+            gapPlusWidthOfBar = ((canvasWidth - rightGap) / noOfBar);
+        gapInXAxix = x = widthOfBar = gapPlusWidthOfBar / 2,
+            rects = [];
+
+        var graphHeight = (canvasHeight * 97) / 100,
+            gapInYaxis = (graphHeight / 10),
+            canvas = document.getElementById('myCanvas'),
+            ctx = canvas.getContext('2d'),
```
Comment (tub99) [inline js/chart.js]: Ctx not to be used here. Use Rectangle.js class

File: js/chart.js [hunk 10]
```diff
@@ -1,22 +1,87 @@
-function Chart(){
-    var renderChart = function(){
-        //code to render charts
-    }
-    this.coloumnChart = function(){
+function Chart() {
+
+    this.coloumnChart = function () {
```
Comment (tub99) [inline js/chart.js]: Chart.js is a class which should contain only a render(type) method. From here u need to make a call to the respective Chart.

File: js/chart.js [hunk 11]
```diff
@@ -1,22 +1,87 @@
...
         return 'yet not decided';
```
Comment (tub99) [inline js/chart.js]: this.render(chartType)  =function() {

// decide which chart object to be created and to render
var x = new XChart();
x.renderChart();
}

File: js/main.js [hunk 12]
```diff
@@ -1,15 +1,38 @@
...
+//function to make canvas responsive
+function CanvasResponsive(canvasId) {
+    var myChartManager,
+        data = [],
+        myParserManager = new parserManager(),
+        myGetData = myParserManager.getData("js/revenue.json");
     myGetData.then(function (resp) {  //try
         myParserManager.myData = resp;
-        var myChartManager = new chartManager(myParserManager.myData);
+        myChartManager = new chartManager(myParserManager.myData);
```
Comment (tub99) [inline js/main.js]: In main.js u can fetch data and only make a call to ChartManager(data, type)

File: js/main.js [hunk 13]
```diff
@@ -1,15 +1,38 @@
...
+        var canvasHeight = 500;
+        if (window.innerWidth > 1000) {
+            canvasWidth = 1000;
+        } else {
+            var canvasWidth = (window.innerWidth * 90) / 100;
+        }
+        canvasHeight = (window.innerHeight * 80) / 100;
+        canvasId.setAttribute("width", canvasWidth);  //setting responsive width to canvas 
+        canvasId.setAttribute("height", canvasHeight);//setting responsive height to canvas 
+        var myAxis = new Axis(canvasId, canvasHeight, canvasWidth, myParserManager.myData);
```
Comment (tub99) [inline js/main.js]: Call to Axis class happens in ChartManger only

File: js/chart.js (hunk 7 above)
Comment (tub99) [inline js/chart.js]: Create a separate function for all calculations and a separate one for rendering

File: js/axis.js [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+            drawMyLine.drawLine(x1, yaxis, canvasWidth, yaxis);
+            myText.fillText(0, yaxis, maxVal, color);
+            maxVal = maxVal - gap;
+            yaxis = yaxis + gapInYaxis;
+        }
+    }
+
+    this.createAxis = function (canvasHeight, canvasWidth, totData) {
+        if (totData.chart.xAxis === 'true') {
+            calXAxis(totData, canvasHeight, canvasWidth);
```
Comment (tub99) [inline js/axis.js]: CalXAxis should not render axis, it should only calculate

File: js/axis.js [hunk 15]
```diff
@@ -0,0 +1,91 @@
+function Axis() {
+    var myText = new GraphText();//creating the object of Text class
+    var drawMyLine = new Line(); //creating the object of Line class
+
+    function calXAxis(totData, canvasHeight, canvasWidth) {
```
Comment (tub99) [inline js/axis.js]: change to  var calXAxis = function() {}

File: js/coloumnChart.js [hunk 16]
```diff
@@ -0,0 +1,102 @@
+var createMyRect = new Rectangle(),//creating the a object of rectangle class
+    tooltip = [];
+
+// function to create coloumnChart
+function ColoumnChart() {
+    Chart.call(this);
+
+    //function to calculate coloumn property. it returns an array of co-ordinates of rectangles 
+    function calculateColomnProprety(totData, canvasWidth, canvasHeight) {
```
Comment (tub99) [inline js/coloumnChart.js:9]: convert pvt function to var yyy = function() not function yyy() {}

**Review by tub99 — APPROVED** [review]
